packages/playwright-core/src/server/chromium/crPage.ts

```diff
@@ -692,6 +692,13 @@ class FrameSession {
       return;
     }
 
+    if (event.targetInfo.type === 'service_worker') {
+      // const work = this._networkManager
+      const worker = this._crPage._browserContext._browser._serviceWorkers.get(event.targetInfo.targetId)!;
+      assert(worker, `Expected to find a worker for ${event.targetInfo.url}`);
```

Comment:
Like this whole MR, there's a lot of unknowns! This assertion is failing a bunch in our trace-viewer tests, so will need to dig in on what assumptions we can/can't make here about timing/ordering of events.